functions/dialogflow/functions/index.js

```diff
@@ -0,0 +1,154 @@
+/* eslint-disable promise/no-nesting */
+/**
+* Copyright 2019 Google LLC
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+ 
+'use strict';
+ 
+const admin = require('firebase-admin');
+const functions = require('firebase-functions');
+const http = require('http');
+const https = require('https');
+admin.initializeApp(functions.config().firebase);
+var db = admin.firestore();
+ 
+// [START save_token_to_firebase]
+function saveOAuthToken(context, oauthToken) {
+  var docRef = db.collection('DialogflowTokens').doc("OauthToken");
```

Comment:
`const` instead of `var`. Since you're using ES6 syntax, I'd always use `let/const` instead of `var`.

Reply:
The functionality of var and let is different from in swift and javascript :) . Updated the code with let/const instead of var.